rules/web/proxy_generic/proxy_susp_ipfs_cred_harvest.yml

```diff
@@ -0,0 +1,22 @@
+title: Suspicious Network Communication With IPFS
+id: eb6c2004-1cef-427f-8885-9042974e5eb6
+status: experimental
+description: Detects connections to interplanetary file system (IPFS) containing a users email address which mirrors behaviours observed in recent phishing campaigns leveraging IPFS to host credential harvesting webpages.
+references:
+    - https://blog.talosintelligence.com/ipfs-abuse/
+    - https://github.com/Cisco-Talos/IOCs/tree/main/2022/11
+    - https://isc.sans.edu/diary/IPFS%20phishing%20and%20the%20need%20for%20correctly%20set%20HTTP%20security%20headers/29638
+author: Gavin Knapp
+date: 2023/03/16
+tags:
+    - attack.credential_access
+    - attack.t1056
+logsource:
+    category: proxy
+detection:
+    selection:
+        cs-uri|re: '(?i)ipfs.io/ipfs.+\..+@.+\..+'
```

Comment:
which of these dots and `\` should be literal and which are used for escaping/matching any character?
I assume at least the dot in `ipfs.io` needs to be escaped

Reply:
Oops apologies, not enough coffee yesterday, hopefully this one is accurate:

(?i)(ipfs\.io/|ipfs\.io\s).+\..+@.+\.[a-zA-Z]+

![image](https://user-images.githubusercontent.com/16122365/225876119-64dd95fb-89c5-421e-b956-f6d14448a350.png)